explore/Classifier_results-different_genes.py

```diff
@@ -0,0 +1,246 @@
+
+# In[1]:
+
+import os
+import urllib
+import random
+import warnings
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import seaborn as sns
+from sklearn import preprocessing, grid_search
+from sklearn.linear_model import SGDClassifier
+from sklearn.cross_validation import train_test_split
+from sklearn.metrics import roc_auc_score, roc_curve
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import StandardScaler
+from sklearn.feature_selection import SelectKBest
+from statsmodels.robust.scale import mad
+
+
+# In[2]:
+
+get_ipython().magic('matplotlib inline')
+plt.style.use('seaborn-notebook')
+
+
+# ## Specify model configuration
+
+# In[3]:
+
+# Parameter Sweep for Hyperparameters
+n_feature_kept = 500
+param_fixed = {
+    'loss': 'log',
+    'penalty': 'elasticnet',
+}
+param_grid = {
+    'alpha': [10 ** x for x in range(-6, 1)],
+    'l1_ratio': [0, 0.05, 0.1, 0.2, 0.5, 0.8, 0.9, 0.95, 1],
+}
+
+
+# *Here is some [documentation](http://scikit-learn.org/stable/modules/generated/sklearn.linear_model.SGDClassifier.html) regarding the classifier and hyperparameters*
+# 
+# *Here is some [information](https://ghr.nlm.nih.gov/gene/TP53) about TP53*
+
+# ## Load Data
+
+# In[4]:
+
+get_ipython().run_cell_magic('time', '', "path = os.path.join('download', 'expression-matrix.tsv.bz2')\nX = pd.read_table(path, index_col=0)")
+
+
+# In[5]:
+
+get_ipython().run_cell_magic('time', '', "path = os.path.join('download', 'mutation-matrix.tsv.bz2')\nY = pd.read_table(path, index_col=0)")
+
+
+# In[6]:
+
+# GENES
+
+genes_LungCancer = {
+    '207': 'AKT1', 
+    '238': 'ALK',  
+    '673': 'BRAF', 
+    '4921':'DDR2',
+    '1956':'EGFR',
+    '2064':'ERBB2',
+    '3845':'KRAS',
+    '5604':'MAP2K1',
+    '4893':'NRAS',
+    '5290':'PIK3CA',
+    '5728':'PTEN',
+    '5979':'RET',
+    '6016':'RIT1',
+    '6098':'ROS1',
+}
+
+genes_TumorSuppressors = {
+    '324': 'APC',  
+    '672': 'BRCA1',  
+    '675': 'BRCA2',
+    '1029':'CDKN2A',
+    '1630':'DCC',
+    '4089':'SMAD4',
+    '4087':'SMAD2',
+    '4221':'MEN1',
+    '4763':'NF1',
+    '4771':'NF2',
+    '7157':'TP53', 
+    '5728':'PTEN', 
+    '5925':'Rb',
+    '7428':'VHL',
+    '7486':'WRN',
+    '7490':'WT1',
+}
+
+list_of_genes = list(genes_LungCancer.keys()) + list(genes_TumorSuppressors.keys())
+
+gene_symbols = list(genes_LungCancer.values()) + list(genes_TumorSuppressors.values())
+
+AUROCS_train = []
+AUROCS_test = []
+p = []
+
+
+# In[7]:
+
+for GENE in list_of_genes:
+    
+    y = Y[GENE]
+    # Here are the percentage of tumors with NF1
+    per = y.value_counts(True)
+    # Set aside 10% of the data for testing, this can only be done where the number of mutations is large enough
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=0)
+    'Size: {:,} features, {:,} training samples, {:,} testing samples'.format(len(X.columns), len(X_train), len(X_test))
+    def fs_mad(x, y):

```

Comment:
You can define `fs_mad` and `get_threshold_metrics` outside of the for loop. Currently, every iteration of the loop redefines the function which is unnecessary because the function does not change.